CONTRIBUTING.md

```diff
@@ -0,0 +1,63 @@
+## Summary of upstream dockerfiles
+
+The kserve server rocks in this repo build images for the upstream dockerfiles located [here](https://github.com/kserve/kserve/tree/master/python).  These server images all have the following common traits:
+
+* they are implemented as python packages and use [poetry](https://python-poetry.org/) to manage their dependencies
+* each server installs its own server-specific package (ex: [sklearn](https://github.com/kserve/kserve/tree/master/python/sklearnserver))
+* they all install a common [kserve](https://github.com/kserve/kserve/tree/master/python/kserve) package
+
+The Dockerfile for each of these images takes advantage of how each server is defined as a poetry package, using `poetry install` in the Dockerfile directly.
+
+## Implementation details of the ROCKs in this repo
+
+The ROCKs for the kserve servers require some atypical workarounds, mostly due to the upstream project using poetry to install its dependencies.  These are documented here in detail, and briefly noted in the rockcraft.yaml files in this repository.
+
+### Installing Python/pip via overlay-packages
+
+We use overlay-packages to get a python3.10 that we can `pip install` packages into.  For some reason if we add python3.10 as a build-package, the python packages we `pip install`/`poetry install` are not added to this python but to a different one.  This problem will first show up when calling `pip install poetry; poetry config ...`, where `pip install poetry` will work fine buy `poetry config ...` will return with command not found.  
+
+As a workaround, we use python/pip from the `overlap-packages`, which somehow makes this work as desired.
+
+### Moving packages installed during the build to the final rock
+
+By listing `python3.10` and `python3-pip` in `overlay-packages`, rockcraft will promote python/pip to the final rock but it **does not automatically migrate any python packages we have installed**.  As a workaround, we copy the installed packages manually by copying the contents of `/usr/local/lib/python3.10/dist-packages` to `$CRAFT_PART_INSTALL`.
```

Comment:
Would rockcraft migrate the packages if we had added python3.10 in build-packages instead?

Reply:
Hmm maybe they are considered as build entities that are used to build an executable that will live in final rock? Anyway, you can resolve this.

Reply:
yeah maybe?  We should ask rockcraft